server/src/gameserver/sdk.rs

```diff
@@ -46,93 +31,92 @@ impl Manager {
+                    BoardEvent::BoardLoaded { id } => self.board_loaded(id).await,
+                    BoardEvent::BoardClosed { id } => self.board_closed(id).await,
                     BoardEvent::UserConnected { board_id, user_id } => {
-                        Self::user_connected(&mut sdk, board_id, user_id, &mut users_by_board).await
+                        self.user_connected(board_id, user_id).await
                     }
                     BoardEvent::UserLeft { board_id, user_id } => {
-                        Self::user_disconnected(&mut sdk, board_id, user_id, &mut users_by_board)
-                            .await
+                        self.user_disconnected(board_id, user_id).await
                     }
                 }
             }
         }
     }
 
-    async fn ready(sdk: &mut Sdk) {
-        sdk.ready()
+    async fn ready(&mut self) {
+        self.sdk
+            .ready()
             .await
             .and({
-                let mut alpha_sdk = sdk.alpha().clone();
+                let mut alpha_sdk = self.sdk.alpha().clone();
                 alpha_sdk.set_player_capacity(1000).await
             })
             .expect("Can't send ready signal.")
     }
 
-    async fn shutdown(sdk: &mut Sdk) {
+    async fn shutdown(&mut self) {
         log::info!("Shutdown server");
-        sdk.shutdown().await.expect("Can't shutdown server.")
+        self.sdk.shutdown().await.or::<()>(Result::Ok(())).unwrap()
     }
 
-    async fn board_closed(sdk: &mut Sdk, space_id: usize, boards_map: &mut BoardByUsers) {
+    async fn board_closed(&mut self, space_id: usize) {
         log::info!("Space id={} disconnected", space_id);
-        boards_map.remove(&space_id);
-        sdk.set_label(format!("board-{}", &space_id), "")
+        self.boards_map.remove(&space_id);
+        self.sdk
+            .set_label(format!("gs-{}", &space_id), "")
             .await
             .expect(format!("Can't set label board-{} -> ''", &space_id).as_str());
 
-        if boards_map.len() == 0 {
-            sdk.shutdown().await.expect("Can't shutdown server.")
+        if self.boards_map.len() == 0 {
+            self.sdk.shutdown().await.expect("Can't shutdown server.")
         } else {
-            sdk.ready() // set ready, we can accept more boards
+            self.sdk
+                .ready() // set ready, we can accept more boards
                 .await
                 .expect("Can't send ready signal.")
         }
     }
 
-    async fn board_loaded(
-        sdk: &mut Sdk,
-        space_id: usize,
-        boards_map: &mut BoardByUsers,
-        max_board_count: usize,
-    ) {
+    async fn board_loaded(&mut self, space_id: usize) {
         log::info!("Space id={} loaded", space_id);
-        boards_map.insert(space_id, HashSet::new());
-        sdk.set_label(format!("gs-{}", space_id), "space-id")
+        self.boards_map.insert(space_id, HashSet::new());
```

Comment:
if multiple invocations with the same space id occur, will this overwrite existing data?

Reply:
good catch - let me fix that real quick

Reply:
done :+1: